sympy/solvers/tests/test_solveset.py

```diff
@@ -932,6 +935,19 @@ def test_issue_9522():
     assert solveset(expr2, x, S.Reals) == EmptySet()
 
 
+def test_solvify():
+    x = Symbol('x')
+
+    assert solvify(x**2 + 10, x, S.Reals) == []
+    assert solvify(x**3 + 1, x, S.Complexes) == [-1, 1/2 - sqrt(3)*I/2,
+                                                 1/2 + sqrt(3)*I/2]
+    assert solvify(log(x), x, S.Reals) == [1]
+    assert solvify(cos(x), x, S.Reals) == [pi/2, 3*pi/2]
+    assert solvify(sin(x) + 1, x, S.Reals) == [3*pi/2]
+    assert solvify(sin(Abs(x)), x, S.Reals) is None
```

Comment:
> So whenever solveset is returning Intersection soln, inequality solvers solve_univariate_inequality will not work. Same for complement soln ? 

Yes.

> Can you write about it in docstring ?

I have mentioned this in `solvify` function.
However, it would be a good idea to state this behaviour in `solve_univariate_inequality` as well.
Thanks ! :)